### PR TITLE
EVG-16681 Update React.FC to be React.VFC

### PR DESCRIPTION
**Pull request by khelif96**
[EVG-16681](https://jira.mongodb.org/browse/EVG-16681)

### Description 
This PR updates our usage of the `React.FC` prop to `React.VFC` which is functionally the same as `React.FC` without the explicit `children` prop.
 @SupaJoon  and I discovered this new type after discussing typing react components without `React.FC` like so
```ts
const SomeComponent = (props: SomePropType) => (...someReturn)
```
While the above example is explicit about the props the component accepts, avoiding a situation in which there is an incorrectly added `children` prop as is the case with `React.FC` it does not have an explicit return type defined.
This can potentially result in a user error in which we accidentally return the wrong thing from a component. 
`React.VFC` solves this issue by not including the unexpected `children` prop while preserving the the return type safety.

I think in general if you are creating a new component that does **require** the `children` prop it is safe to use either `React.FC` or `React.VFC` and explicitly type `children`. 

This is all subject to change, there is supposedly the plan to update `React.FC` to have the same behavior as `React.VFC` by default in the next major release. (React 18) 
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46643#issuecomment-801487166

But since we will likely not be upgrading to React 18 in the near future i think it's ok to make this change for now and update it when we do.

Read more here. (Yes the naming sucks)
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46643

### CODE REVIEW

File: src/components/DisplayModal.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ interface DisplayModalProps {
   ) => void | React.Dispatch<React.SetStateAction<boolean>>;
   size?: ModalSize;
   title?: string;
+  children: React.ReactNode;
 }
 
 export const DisplayModal: React.FC<DisplayModalProps> = ({
```
Comment (sophstad) [inline src/components/DisplayModal.tsx:18]:
```suggestion
export const DisplayModal: React.VFC<DisplayModalProps> = ({
```

File: src/context/Providers.tsx [hunk 2]
```diff
@@ -3,7 +3,9 @@ import LeafyGreenProvider from "@leafygreen-ui/leafygreen-provider";
 import { AuthProvider } from "context/auth";
 import { ToastProvider } from "context/toast";
 
-export const ContextProviders: React.FC = ({ children }) => (
+export const ContextProviders: React.FC<{ children: React.ReactNode }> = ({
```
Comment (sophstad) [inline src/context/Providers.tsx:6]:
```suggestion
export const ContextProviders: React.VFC<{ children: React.ReactNode }> = ({
```

**Review by sophstad — APPROVED** [review]
LGTM once all remaining instances of `React.FC` are converted into `React.VFC` 👍 thanks!!

File: src/components/Spawn/Layout.tsx [hunk 3]
```diff
@@ -22,18 +23,25 @@ export const StyledBadge = styled(Badge)`
   margin: 0 ${size.xs};
 `;
 
-export const PlusButton = ({ children, ...props }) => (
-  <Button {...{ ...props, glyph: <Icon glyph="Plus" /> }} as="button">
-    {children}
-  </Button>
-);
+export const PlusButton: ExtendableBox<
```
Comment (khelif96) [inline src/components/Spawn/Layout.tsx:26]: This button was never actually typed correctly so it broke when we updated the Leafygreen Button and the props changed.

**Review by SupaJoon — APPROVED** [review]

**Pull request merged**